crates/subspace-farmer/src/farmer_cache.rs

```diff
@@ -488,42 +453,31 @@ where
             };
 
             // Find plot in which there is a place for new piece to be stored
-            let mut sorted_caches = caches.iter_mut().enumerate().collect::<Vec<_>>();
-            // Sort piece caches by number of stored pieces to fill those that are less
-            // populated first
-            sorted_caches.sort_by_key(|(_, cache)| cache.stored_pieces.len());
-            if !stream::iter(sorted_caches)
-                .any(|(cache_index, cache)| async move {
-                    let Some(offset) = cache.free_offsets.pop_front() else {
-                        return false;
-                    };
+            let Some(offset) = caches.free_offsets.pop_front() else {
```

Comment:
This will work, but it loses the property where pieces were distributed across multiple caches for performance reasons. Unless offsets interleave this will likely result in higher read load on one disk and lower on another, which may negatively impact farming performance.

Reply:
Let me think about it, I should somehow make the cache load as balanced as possible, and I think we can do that.